File: src/Lession4/Ex3.java
package Lession4;

import java.util.Scanner;

public class Ex3 {
    static void main() {
        Scanner sc = new Scanner(System.in);
        String forward = sc.next();
        String backward = sc.next();

        if (checkReverse(forward, backward)) {
            System.out.println("YES");
        } else {
            System.out.println("NO");
        }
    }

    static boolean checkReverse (String strA, String strB) {
        // Check lengths
        if (strA.length() != strB.length()) {
            return false;
        }

        // Check reverse
        boolean res = true;
        for (int k = 0; k < strA.length(); k++) {
            if (strA.charAt(k) != strB.charAt(strA.length() - k - 1)) {
                res = false;
                break;
            }
        }

        return res;
    }
}

File: src/Lession4/Ex3.kt
package Lession4

fun main() {
    val forward = readln();
    val backward = readln();

    if (checkReverse(forward, backward)) {
        println("YES")
    } else {
        println("NO")
    }
}

fun checkReverse (strA: String, strB: String): Boolean {
    if (strA.length != strB.length) {
        return false
    }

    var res = true;
    for (i in 0 until strA.length) {
        if (strA[i] != strB[strA.length - i - 1]) {
            res = false
            break;
        }
    }

    return res
}
